homeassistant/components/neato/camera.py

```diff
@@ -102,33 +109,33 @@ def update(self):
 
         self._image = image.read()
         self._image_url = image_url
-        self._generated_at = map_data["generated_at"]
+        self._generated_at = map_data.get("generated_at")
         self._available = True
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of this camera."""
         return self._robot_name
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return unique ID."""
         return self._robot_serial
 
     @property
-    def available(self):
+    def available(self) -> bool:
         """Return if the robot is available."""
         return self._available
 
     @property
-    def device_info(self):
+    def device_info(self) -> DeviceInfo:
         """Device info for neato robot."""
         return {"identifiers": {(NEATO_DOMAIN, self._robot_serial)}}
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes of the vacuum cleaner."""
-        data = {}
+        data: dict = {}
```

Comment:
This may need a more detailed type to correspond to the return type of the property.

Reply:
Fixed